forwardproxy.go

```diff
@@ -695,8 +701,184 @@ type dialContexter interface {
 	DialContext(ctx context.Context, network, address string) (net.Conn, error)
 }
 
+func (h *Handler) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	if !d.Next() {
+		return d.ArgErr()
+	}
+	args := d.RemainingArgs()
+	if len(args) > 0 {
+		return d.ArgErr()
+	}
+	for nesting := d.Nesting(); d.NextBlock(nesting); {
+		subdirective := d.Val()
+		args := d.RemainingArgs()
+		switch subdirective {
+		case "basicauth":
```

Comment:
```suggestion
		case "basic_auth":
```

Reply:
Do we want to be compatible with existing configs and docs?

Reply:
No, Caddy 2 is not backwards compatible with Caddy 1.